### ISSUE TITLE
NotImplementedError: fork() on Windows

### ISSUE THREAD

**Issue by hyrious** [piotrmurach/tty-pager#7]
~~~
Ruby/lib/ruby/gems/2.3.0/gems/tty-pager-0.7.1/lib/tty/pager/system.rb:62:in `fork'
~~~

Briefly, it's not friendly to see the error. :thinking:

**Comment by piotrmurach** [comment]
Thanks for giving this library a try!

Would mind providing more information on the type of console you're using on Windows?

The `TTY::Pager` should automatically switch between `SystemPager` and `BasicPager`. However, this seems to fail in this situation as probably a given paging utility exists on your system that passes the check but you're probably running vanilla powershell without forking capability? For the time being, you should be able to use `TTY::Pager::BasicPager` directly to paginate. 

Do you have time to look into why switching to `BasicPager` fails on your system?

**Comment by hyrious** [comment]
It turns out that `TTY::Pager::SystemPager.available?` returns `true`.

**TL;DR :** the command `more` is executable on Windows, which causes the `TTY::Pager` switch to `SystemPager`.

**However**, it's not safe to check environment by this way, because I usually use these unix-like tools from mingw or git-for-windows.

~~~ cmd
C:\Users\hyrious>more /?
Displays output one screen at a time.

MORE [/E [/C] [/P] [/S] [/Tn] [+n]] < [drive:][path]filename
command-name | MORE [/E [/C] [/P] [/S] [/Tn] [+n]]
MORE /E [/C] [/P] [/S] [/Tn] [+n] [files]
~~~

May be check the RUBY_PLATFORM will be a better way.
- - -

I looked into source code, the `SystemPager` will check executables of
~~~ ruby
[ENV['GIT_PAGER'], ENV['PAGER'],
 `git config --get-all core.pager`,
 'less', 'more', 'cat', 'pager']
~~~
which in my circumstance is
~~~ ruby
[nil, nil, "", 'less', 'more', 'cat', 'pager']
~~~
after `.compact.map(&:strip).reject(&:empty?).uniq.map { |cmd| TTY::Which.exist? cmd }`, there are
~~~ ruby
[true, true, true, false] # the ENV includes git-for-windows, as in Cmder
[false, true, false, false] # `more` is executable on windows' cmd
~~~

**Comment by piotrmurach** [comment]
Thanks for looking into this. As suspected the `available?` is broken and this confirms it. Checking for platform is less than optimal as people may well be using `cygwin` which actually supports forking behaviour on Windows. I will see if I can fix it later today. Will you have time to test the behaviour from master? If you have any ideas beforehand please submit PR and I will happily review.

**Comment by piotrmurach** [comment]
@hyrious I have used your suggestion to detect fork presence on Windows. I've also added friendly error message that prevents initializing `SystemPager` if it cannot be used. Would you mind testing master branch?

**Comment by hyrious** [comment]
Thank you! It works fine now.
